test/feature/AsyncGenerators/resources/observable.js

```diff
@@ -0,0 +1,71 @@
+const generator_ = Symbol();
+const onDone_ = Symbol();
+const generatorFunction_ = Symbol();
+
+function schedule(asyncF) {
+  return Promise.resolve().then(asyncF);
+}
+
+class DecoratedGenerator {
+  constructor(generator, onDone) {
+    this[generator_] = generator;
+    this[onDone_] = onDone;
+  }
+
+  next(value) {
+    var result = this[generator_].next();
+    if (result.done) {
+      this[onDone_]();
```

Comment:
You mean onDone.call(?, ...)? I don't know of a good value for ? here. Could be this, the instance of DecoratedGenerator. But in any case, this is only part of a test, and that part should later be replaced by a proper async generator.